src/rail/estimation/algos/deepdisc.py

```diff
@@ -358,7 +365,62 @@ def finalize(self):
         self.add_handle("output", data=qp_distn)
         truth_dict = dict(redshift=self.true_zs)
         self.add_handle("truth", data=truth_dict)
+'''
+
+def _do_inference(q, predictor, metadata, num_gpus, batch_size, zgrid):
+        """This is the function that is called by `launch` to parallelize
+        inference across all available GPUs."""
+
+        group = dist.new_group()
 
+        mapper = RedshiftDictMapperEval(
+            DC2ImageReader(), lambda dataset_dict: dataset_dict["filename"]
+        ).map_data
+
+        loader = d2data.build_detection_test_loader(
+            metadata, mapper=mapper, batch_size=batch_size
+        )
+
+        # this batched version will break up the metadata across GPUs under the hood.
+        true_zs, pdfs, ids = run_batched_match_redshift(loader, predictor, ids=True)
```

Comment:
It would be useful to make the ids a stage config dict option, although this can be a separate PR

Reply:
Done. I've added a new boolean stage parameter called `return_ids_with_inference` (the name may not be ideal) and piped it through to _do_inference. 

Since run_batched_match_redshift will return `[]` if ids=False, we can let the logic remain as is, and only introduce conditional logic to control whether or not we try to add `all_ids` to the ancillary data of the qp ensemble.